cmd/up/exec_test.go

```diff
@@ -324,6 +324,63 @@ func TestGetEnvsError(t *testing.T) {
 	}
 }
 
+func TestGetEnvForHybridModeOverwrittesFromManifestDeclaration(t *testing.T) {
+	ctx := context.Background()
+
+	fakeConfigMapEnvsGetter := fakeGetter{envs: []string{"ENVTOOVERWRITE=FROMCONFIGMAP"}}
+	fakeSecretEnvsGetter := fakeGetter{envs: []string{"ENVTOOVERWRITE=FROMSECRET"}}
+	fakeImageEnvsGetter := fakeGetter{envs: []string{"ENVTOOVERWRITE=FROMIMAGE"}}
+	client := fake.NewSimpleClientset(&appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "test",
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Env: []v1.EnvVar{
+								{
+									Name:  "ENVTOOVERWRITE",
+									Value: "FROMPOD",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	})
+	dev := &model.Dev{
+		Name:      "test",
+		Namespace: "test",
+		Environment: model.Environment{
+			model.EnvVar{
+				Name:  "ENVTOOVERWRITE",
+				Value: "FROMMANIFEST",
+			},
+		},
+	}
+	eg := envsGetter{
+		dev:                 dev,
+		name:                "test",
+		namespace:           "test",
+		client:              client,
+		configMapEnvsGetter: &fakeConfigMapEnvsGetter,
+		secretsEnvsGetter:   &fakeSecretEnvsGetter,
+		imageEnvsGetter:     &fakeImageEnvsGetter,
+		getDefaultLocalEnvs: func() []string { return []string{} },
+	}
+	envs, err := eg.getEnvs(ctx)
+	require.NoError(t, err)
+
+	// following docs, if exec.cmd.Env contains duplicate environment keys, only the last value in the
+	// slice for each duplicate key is used so we can check overwrittes by getting last value
+	// in envs list.
+	require.Equal(t, "ENVTOOVERWRITE=FROMMANIFEST", envs[len(envs)-1])
```

Comment:
Could we ensure that the order is the correct instead of checking just the last one? That way we also cover the other cases.

Reply:
checking your suggestions I think we don't have any preferred order except those coming from the manifest, so a test checking the order of all of them seems strange to me. I think that with the test we have now we cover the only preference we have. WDYT?

Reply:
@pchico83 What should be the order of priority between the shipments coming from the configmap, secrets, image pod and manifest?

Reply:
My 2¢ is (being 1 the top priority):
1. manifest
2. pod
3. configmap
4. secrets
5. image
But let's double check before going with an approach

Reply:
+1

Reply:
4fbd68b96a114bb496c9cc95116a387ba43cde32